plugins/outputs/sumologic/sumologic_test.go

```diff
@@ -592,26 +603,35 @@ func TestMaxRequestBodySize(t *testing.T) {
 				s.MaxRequstBodySize = 300
 				return s
 			},
-			metrics:              getMetrics(t, count),
-			expectedError:        false,
-			expectedRequestCount: 100,
+			metrics:                  getMetrics(t, count),
+			expectedError:            false,
+			expectedRequestCount:     100,
+			expectedMetricLinesCount: 500, // count (100) metrics, 5 lines per each (steal, idle, system, user, temp) = 500
 		},
 	}
 
 	for _, tt := range testcases {
 		t.Run(tt.name, func(t *testing.T) {
+			var (
+				requestCount int
+				linesCount   int
+			)
+			ts.Config.Handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				requestCount++
```

Comment:
might need `atomic.AddInt32` / `atomic.LoadInt32` for these counters. It's a different goroutine, and isn't race safe.